### ISSUE TITLE
Develop a sustainable solution for using workload identity

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#13]
### TL;DR

The design outlined in #11 is problematic because it gives broad permissions to the CSI Driver `DaemonSet`, specifically `serviceaccounts/token`. This may break/conflict with the users' ability to use [node isolation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#node-isolation-restriction) because the `serviceaccounts/token` permission is not attenuated based on the node.

### Design

**Proposal**

TBD. Investigate alternatives that do not require granting `serviceaccounts/token` to the driver and ensure the solution does not conflict with node isolation.

**Alternatives considered**

* https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/issues/11
* The CSI Driver could mount the kubelet config, which would give it node-permissions, but this does not seem like a stable pattern to follow and could break in the future.

**Resources**

- https://github.com/kubernetes/kubernetes/issues/86448

**Additional information**

**Comment by tam7t** [comment]
`CSIServiceAccountToken` is Alpha in 1.20 (https://github.com/kubernetes/kubernetes/pull/93130)

**Comment by tam7t** [comment]
Now that https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/805 is merged, the provider should use the K8s SA token if one is provided. For configurations that use 805 an SA token should always be present in the mount request.

As https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/585 get resolved in the driver then we can start to always assume that a SA token is present, but that will require [k8s 1.22](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1855-csi-driver-service-account-token/README.md#implementation-history). 1.21 is EOL [2022-06-28](https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md#121).

**Comment by amitmodak** [comment]
Changes have been merged via https://github.com/GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp/pull/274.
